elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/ElasticsearchQueries.java

```diff
@@ -46,23 +53,18 @@ public static QueryDescriptor fromQueryString(String queryStringQuery) {
 		// Payload looks like so:
 		// { "query" : { "query_string" : { "query" : "abstract:Hibernate" } } }
 
-		JsonObject query = new JsonObject();
-		query.addProperty( "query", queryStringQuery );
-
-		JsonObject queryString = new JsonObject();
-		queryString.add( "query_string", query );
-
-		JsonObject queryObject = new JsonObject();
-		queryObject.add( "query", queryString );
+		JsonBuilder.Object query = JsonBuilder.object().add( "query",
+				JsonBuilder.object().add( "queryString",
+						JsonBuilder.object().addProperty( "query", queryStringQuery ) ) );
 
-		return new ElasticsearchJsonQuery( queryObject.toString() );
+		return new ElasticsearchJsonQuery( query.build() );
 	}
 
 	private static class ElasticsearchJsonQuery implements QueryDescriptor {
 
-		private final String jsonQuery;
+		private final JsonObject jsonQuery;
 
-		public ElasticsearchJsonQuery(String jsonQuery) {
+		public ElasticsearchJsonQuery(JsonObject jsonQuery) {
```

Comment:
Hi, as we discussed I'd prefer to not expose the usage of Gson on the public API at this point. This should be an implementation detail only.